### PR TITLE
YOLOv5 -- 45328107 -- xspinella

### PR DESCRIPTION
**Pull request by LeoCain**
To the Patternflow team,
This pull request presents implementation of the YOLOv5 module on the ISIC-2017 dataset. The problem has been defined as detection and classsification of lesions as either "melanoma" or "not melanoma." The medium-sized model (YOLOv5m) was used, and the performance on the test dataset was as follows:

mAP:0.5-0.95 = 46.6%
Average IOU = 0.814
Classification accuracy = 0.868

While this may seem to be a success, further investigation found the good classification accuracy was mainly a result of the model classifying negative melanoma really well, however, its classification of positive melanoma cases was quite poor as shown by the precision and recall measures below, which were computed in regards to detection of positive melanoma cases:

P = TP/(TP+FP) = 39/39+21 -> "~65% of positive predictions are correct"
R = TP/(TP+FN) = 39/39+44 -> "~46% of positive cases are diagnosed successfully"

So why were the mAP, IOU, and class accuracy acceptable, but positive detections poor? This is due to the distribution on the dataset, which was largely skewed towards negative diagnoses, as one would expect with medical data. Please view a more elaborated discussion of this, including recommendations for improvements, in the README of this pull request.
Regards,
Xavier Spinella -- 45328107

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
- Commit messages are good.
- Readme is excellent! Very descriptive with extensive discussion on all experiments performed, and plots provided for all metrics.
- Performance is good.
- Although using pretrained ultralytics/YOLOv5, has performed hyperparameter tuning beyond the default settings
- Everything looks OK

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove weights and model files -2
Request Description good

**Comment by LeoCain** [comment]
Thank you for the feedback,
Weight/model files have been removed as required.
Let me know if there are any other issues.

**Comment by shakes76** [comment]
Conflicting .gitmodules file needs resolving before merge, feedback marks will be awarded. To get merged though fix conflict.

**Comment by LeoCain** [comment]
It appears that other students have been able to somehow merge their version of .gitmodules, but I am unable to:
image
as you can see, my version specifies my module, but the current master branch specifies modules for some other students. The only way I can see of fixing this from my end would be to delete the submodule specifications from my .gitmodules file, but I am unsure if this will break my code. I will do it anyway, but just thought this was worth mentioning.

**Comment by LeoCain** [comment]
Merge conflict has been corrected as per my previous comment. It appears that with the "path" and "url" specifications missing for my submodule, the "git submodule init" and "git submodule update" commands may be broken, but really I am unsure. The photo in my previous comment shows what the .gitmodules file needs to contain, if this change does happen to break the code.
Please let me know if there are any issues.
-Xavier
